draft-ietf-taps-interface.md

```diff
@@ -1811,6 +1832,37 @@ that the other endpoint is done sending on a connection.
 
 Any calls to Receive once the Final Message has been delivered will result in errors.
 
+
+# Message Contexts {#msg-ctx}
+
+Using the MessageContext object, the application can set and retrieve meta-data of the message, including Message Properties (see {{message-props}}) and framing meta-data (see {{framing-meta}}). 
+Therefore, a MessageContext object can be passed to the Send action and is retuned by each Send and Receive related events.
+
+Message properties can be set and queried using the Message Context:
+
+~~~
+MessageContext.add(scope?, parameter, value)
+PropertyValue := MessageContext.get(scope?, property)
+~~~
+
+To get or set Message Properties, the optional scope parameter is left empty, for framing meta-data, the framer is passed.
+
+The application can also query information about the local and remote endpoint.
+
+~~~
+RemoteEndpoint := MessageContext.GetRemoteEndpoint()
+LocalEndpoint := MessageContext.GetLocalEndpoint()
```

Comment:
Are these only queryable on received messages?

Reply:
I'd say they should be queryable on any MessageContext, but for sent messages, this is currently only possible after send. I'll add a note for this.